substrate/frame/balances/src/lib.rs

```diff
@@ -911,8 +911,8 @@ pub mod pallet {
 			let result = T::AccountStore::try_mutate_exists(who, |maybe_account| {
 				let is_new = maybe_account.is_none();
 				let mut account = maybe_account.take().unwrap_or_default();
-				let did_provide =
-					account.free >= Self::ed() && Self::have_providers_or_no_zero_ed(who);
+				let did_provide = (account.reserved + account.free) >= Self::ed() &&
```

Comment:
Just to confirm: so the `ed` can be part of `reserved`?

Reply:
With the breaking currency api update you now always need to hold at least the ed as free balance. 
That was not the case before. Contracts accounts for example would typically have some amount reserved as deposit and hold 0 balance. 

The code touched here does not take this scenario into account, and migrating these old accounts will wrongly bump the providers count 1. 

The changes here attempt to fix it. I need to look at these failing ci jobs though

Reply:
I see, thanks. Let's also add a comment to explain.